plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java

```diff
@@ -70,6 +77,69 @@ protected boolean isPopulateClientAppName() {
         return false;
     }
 
+    @Override
+    protected Properties getAllConnectionProperties(
+        @NotNull DBRProgressMonitor monitor,
+        JDBCExecutionContext context,
+        String purpose,
+        DBPConnectionConfiguration connectionInfo
+    ) throws DBCException {
+        Properties props = super.getAllConnectionProperties(monitor, context, purpose, connectionInfo);
+        DBPPreferenceStore prefStore = getContainer().getPreferenceStore();
+        if (!prefStore.getBoolean(ModelPreferences.META_CLIENT_NAME_DISABLE)) {
+            String appName;
+            if (prefStore.getBoolean(ModelPreferences.META_CLIENT_NAME_OVERRIDE)) {
+                appName = prefStore.getString(ModelPreferences.META_CLIENT_NAME_VALUE);
+            } else {
+                appName = getPartnerApplicationIdentifier(GeneralUtils.getProductName());
+            }
+            props.put(APPLICATION_PROPERTY, appName);
+        }
+        return props;
+    }
+
+    /**
+     * Converts product name to the application identifier <br/>
+     * <br/>
+     * DBeaver <br/>
+     *     CE = DBeaver_Community <br/>
+     *     EE = DBeaver_Enterprise <br/>
+     *     UE = DBeaver_Ultimate <br/>
+     *     LE = DBeaver_Lite <br/>
+     *     TE = DBeaver_Team <br/>
+     * <br/>
+     * CloudBeaver<br/>
+     *     EE = DBeaver_CloudBeaverEnterprise <br/>
+     *     TE = DBeaver_Team <br/>
+     *     CB AWS = DBeaver_CloudBeaverAWS <br/>
+     * <br/>
+     */
+    private String getPartnerApplicationIdentifier(String productName) {
```

Comment:
Let's not use particular product identitfier here. 
Use DBUtils.getClientApplicationName
Just replace restricted characters with underscore

Reply:
We need specific values that can be fixed